### DISCUSSION TITLE
Reducing timeouts?

### DISCUSSION THREAD

**Discussion by IoTrees** [miguelgrinberg/microdot#181] [Q&A]
Hello,
This is a bit of a javascript / web design question, and might not best the best for this forum, but figured I would ask:

I have an async microdot server running on a WIZNET W5100S-EVB-Pico board, serving a page with inline javascipt that does an XMLHTTPRequest for some JSON data from the server. It does respond and work correctly, but I am getting a lot of timeouts, does anyone have any general advice on reducing timeouts?

```
             function update_readings() {
                 let xhttp = new XMLHttpRequest();
                 console.log("UR intervalId: ",intervalID)
                 xhttp.onreadystatechange = function () {
                     if (this.readyState === 4 && this.status === 200) {

                         JSON PARSING HAPPENS HERE

                 }
                 xhttp.open("GET", "send_data", true);
                 xhttp.send();
                 // xhttp.timeout = 10000;
                 xhttp.ontimeout = function (){
                     console.log("Timeout")
                     document.getElementById("timeout_window").className = "visible-error";
                 }
             }

             let intervalID = setInterval(update_readings,1000)
```

Here is my python function for returning response:

```
@app.get('/send_data')
def send_data(request):
    print("send_data")
    # return "a"
    return await sensor_data()
```

**Comment by miguelgrinberg** [comment]
Is your `sensor_data` function designed to not block the loop? The most likely cause of the timeouts is that your application blocks.
